partiql-planner/src/test/kotlin/org/partiql/planner/TyperTest.kt

```diff
@@ -0,0 +1,236 @@
+package org.partiql.planner
+
+import com.amazon.ionelement.api.loadSingleElement
+import org.partiql.parser.PartiQLParser
+import org.partiql.plan.Statement
+import org.partiql.plan.debug.PlanPrinter
+import org.partiql.planner.catalog.Catalog
+import org.partiql.planner.catalog.Name
+import org.partiql.planner.catalog.Session
+import org.partiql.planner.internal.TestCatalog
+import org.partiql.planner.test.Test
+import org.partiql.planner.util.ProblemCollector
+import org.partiql.plugins.local.toStaticType
+import org.partiql.types.PType
+
+/**
+ * This is a port of the original assertion logic from [org.partiql.planner.internal.typer.PlanTyperTestsPorted].
+ *
+ * @see Success
+ * @see org.partiql.planner.internal.typer.PlanTyperTestsPorted.TestCase.SuccessTestCase
+ * @see Failure
+ * @see org.partiql.planner.internal.typer.PlanTyperTestsPorted.TestCase.ErrorTestCase
+ */
+abstract class TyperTest(
+    private val _name: String,
+    open var expectedType: PType?,
+    var statement: String,
+    var catalog: String,
+    var directory: List<String>,
+) : Test {
+
+    val parser = PartiQLParser.default()
+    val planner = PartiQLPlanner.builder().signal().build()
+
+    override fun getName(): String {
+        return _name
+    }
+
+    override fun toString(): String {
+        return getName()
+    }
+
+    val catalogs: List<Catalog> by lazy {
+        // Make a map from catalog name to tables.
+        val inputStream = this::class.java.getResourceAsStream("/resource_path.txt")!!
+        val map = mutableMapOf<String, MutableList<Pair<Name, PType>>>()
+        inputStream.reader().readLines().forEach { path ->
+            if (path.startsWith("catalogs/default")) {
+                val schema = this::class.java.getResourceAsStream("/$path")!!
+                val ion = loadSingleElement(schema.reader().readText())
+                val staticType = ion.toStaticType()
+                val steps = path.substring(0, path.length - 4).split('/').drop(2) // drop the catalogs/default
+                val catalogName = steps.first()
+                // args
+                val name = Name.of(steps.drop(1))
+                val ptype = PType.fromStaticType(staticType)
+                if (map.containsKey(catalogName)) {
+                    map[catalogName]!!.add(name to ptype)
+                } else {
+                    map[catalogName] = mutableListOf(name to ptype)
+                }
+            }
+        }
+        // Make a catalogs list
+        map.map { (catalog, tables) ->
+            TestCatalog.builder()
+                .name(catalog)
+                .apply {
+                    for ((name, schema) in tables) {
+                        createTable(name, schema)
+                    }
+                }
+                .build()
+        }
+    }
+
+    /**
```

Comment:
Could provide the description of the test format in some other more visible location such as https://github.com/johnedquinn/partiql-lang-jvm/blob/v1-tests-dump/partiql-planner/src/testFixtures/resources/README.adoc.

Something like https://github.com/partiql/partiql-tests/blob/main/docs/partiql-tests-schema-proposal.md#evaluation-tests, could make it easier to add future tests.